src/util/compareRanks.ts

```diff
@@ -0,0 +1,23 @@
+/** Compare ranks between two thoughts. */
```

Comment:
Is there a reason for the complexity in this function, where we convert the ranks to lists of numbers (separated by the decimal point) and compare those individually?

The function `compareRanks` is only used in one place to check for a negative number (`compareRanks(newRank, currentRank) < 0`), and it seems to return a negative number if and only if `rankA < rankB`. I might be misunderstanding some edge cases that this handles; can you please explain why we can't compare the ranks simply as `rankA - rankB`?

Reply:
I was not too sure if there could be ranks like 1.1.2, 1.1.3 etc, but given the possibility I had the function written like this. It certainly can be done with just `rankA - rankB`, if there are not any ranks as above, please do correct me if I am wrong here.

Reply:
Since the parameters are of type `number`, values such as `1.1.2` would not be possible.

Reply:
Hmm, not sure why I assumed it was string, but in that case we can omit the `compareRanks` function and simple do `newRank - currentRank` and this would work just fine.